src/V3ThreadPool.cpp

```diff
@@ -0,0 +1,83 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Thread pool for Verilator itself
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2005-2022 by Wilson Snyder.  This program is free software; you can
+// redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+//
+// Verilator is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+//*************************************************************************
+
+#include "config_build.h"
+
+#include "V3ThreadPool.h"
+
+#include "V3Ast.h"
+#include "V3Error.h"
+
+// The global thread pool
+V3ThreadPool v3ThreadPool;
+
+void V3ThreadPool::resize(unsigned n) {
+    UASSERT(m_queue.empty(), "Resizing busy thread pool");
+    // Shut down old threads
+    m_shutdown = true;
+    m_cv.notify_all();
+    while (!m_workers.empty()) {
+        m_workers.front().join();
+        m_workers.pop_front();
+    }
+    // Start new threads
+    m_shutdown = false;
+    for (unsigned int i = 0; i < n; i++) {
+        m_workers.emplace_back(&V3ThreadPool::worker, this, i);
```

Comment:
I’m not sure this is well defined (I know I wrote it), please add a static startWorker function that explicitly takes a V3ThreadPool pointer (selfp) and the id, and then just calls selfp->worker(id)